src/sentry/issues/endpoints/group_reprocessing.py

```diff
@@ -20,12 +24,30 @@ def post(self, request: Request, group) -> Response:
 
         This endpoint triggers reprocessing for all events in a group.
 
+        Returns a 409 if the group itself is currently on the receiving-end
+        of a reprocessing.
+
         :pparam string issue_id: the numeric ID of the issue to reprocess. The
             reprocessed events will be assigned to a new numeric ID. See comments
             in sentry.reprocessing2.
         :auth: required
         """
 
+        # Using raw SQL since data is LegacyTextJSONField which can't be filtered with Django ORM
+        parent = (
+            Activity.objects.annotate(new_group_id=RawSQL("(data::json ->> 'newGroupId')", []))
+            .filter(type=ActivityType.REPROCESS.value, new_group_id=str(group.id))
+            .order_by("-datetime")
+            .first()
+        )
+        if parent and not is_group_finished(parent.group_id):
+            return self.respond(
+                {
+                    "error": "group is part of an ongoing reprocessing",
+                },
+                status=409,
```

Comment:
We might want to improve the front-end for this in a separate front-end PR to let the user know why the reprocessing failed 🤔